File: src/main/kotlin/com/github/frcsty/frozenjoin/action/util/UtilArray.java
package com.github.frcsty.frozenjoin.action.util;

public final class UtilArray {

    public static Object[] prepend(final Object[] array, final Object value) {
        final Object[] newArray = new Object[array.length + 1];

        newArray[0] = value;
        System.arraycopy(array, 0, newArray, 1, array.length);

        return newArray;
    }
}

File: src/main/kotlin/com/github/frcsty/frozenjoin/action/util/UtilArray.kt
package com.github.frcsty.frozenjoin.action.util

object UtilArray {
    fun prepend(array: Array<Any?>, value: Any?): Array<Any?> {
        val newArray = arrayOfNulls<Any>(array.size + 1)
        newArray[0] = value
        System.arraycopy(array, 0, newArray, 1, array.size)
        return newArray
    }
}